src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java

```diff
@@ -101,20 +106,25 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
     }
 
     private List<Testsuite> extractTestResults(@Nonnull TaskListener taskListener, FilePath resultsDir) throws IOException, InterruptedException {
-        return resultsDir.list().stream()
-                .filter(path -> path.getName().endsWith(".xml"))
-                .map(report -> {
-                    try {
-                        final JAXBContext context = JAXBContext.newInstance(Testsuite.class);
-                        final Unmarshaller unmarshaller = context.createUnmarshaller();
-                        return (Testsuite) unmarshaller.unmarshal(report.read());
-                    }
-                    catch (JAXBException | IOException | InterruptedException e) {
-                        taskListener.error(e.getMessage(), e);
-                        throw new TestParsingException(e);
-                    }
-                })
-                .collect(Collectors.toList());
+        return resultsDir.list().stream().filter(path -> path.getName().endsWith(".xml")).map(report -> {
+            try {
+                final JAXBContext context = createJAXBContext();
+                final Unmarshaller unmarshaller = context.createUnmarshaller();
+                return (Testsuite) unmarshaller.unmarshal(report.read());
+            }
+            catch (JAXBException | IOException | InterruptedException e) {
+                taskListener.error(e.getMessage(), e);
+                throw new TestParsingException(e);
+            }
+        }).collect(Collectors.toList());
+    }
+
+    private JAXBContext createJAXBContext() throws JAXBException {
+        return ContextFactory.createContext( //
+                ObjectFactory.class.getPackage().getName(), //
+                ObjectFactory.class.getClassLoader(), //
+                null //
+        );
```

Comment:
How would this part look like if you wouldn’t force the line breaks with the comments?

Reply:
Would be a single long line ..

```
    private JAXBContext createJAXBContext() throws JAXBException {
        return ContextFactory.createContext(ObjectFactory.class.getPackage().getName(), ObjectFactory.class.getClassLoader(), null);
    }
```

Reply:
I do not have a strong opinion here. Personally, I think the upper one is better to read, but that's just a personal opinion :)

Reply:
I’m not a fan of long lines either, but as long as spotless doesn’t just add an arbitrary newline in the middle of the arguments list it at least fits in with the rest of the code style.

Reply:
In general, you are right. I'll push it :)